src/disposable.coffee

```diff
@@ -24,7 +24,9 @@ class Disposable
   dispose: ->
     unless @disposed
       @disposed = true
-      @disposalAction?()
```

Comment:
Personally, I don't think it's good for `Disposable` to accept something that is not a function and silently ignore it because that's likely masking a bug, but I'll preserve the existing behavior.

I'm suspicious that if this were not written in CoffeeScript, no such concession would have been made :P